File: Python/sorting.py
import time
from Quick import QuickSort

def main():
    # Import list of text to sort
    file = open("sgb-words.txt")
    data = file.read().split("\n")
    file.close()

    # Timestamp when sorting starts
    start = time.time()

    # Quick Sort
    QuickSort(data, 0, len(data)-1)
    
    # Timestamp when sorting ends
    end = time.time()

    # Output execution time
    print ("Time elapsed: " , int((end-start)*1000), "ms")
    
    # Record sorting result in text file called "sgb-words-sorted-py.txt"
    file = open("Python/sgb-words-sorted-py.txt", "w")
    for i in data:
        file.write(i)
        if i==data[len(data)-1]:
            break
        file.write("\n")
    file.close()

if __name__ == "__main__":
    main()

File: Java/sorting.java
package Java;
import static Java.algo.Quick.*;
import static Java.utility.utility.*;
import java.io.IOException;

public class sorting {
    public static void main(String[] args) throws IOException{
        // Import list of text to sort
        String[] data = ImportFile("sgb-words.txt");

        // Timestamp when sorting starts
        long start = System.nanoTime();
        
        // Quick Sort
        QuickSort(data, 0, data.length-1);

        // Timestamp when sorting ends
        long end = System.nanoTime();

        // Output execution time
        System.out.println("Time executed: " + (end - start)/1000000 + " ms");

        // Record sorting result in text file called "sgb-words-sorted-java.txt"
        ExportFile("Java/sgb-words-sorted-java.txt", data);
    }
}
